record-and-playback/core/scripts/post_publish/export_slides.rb

```diff
@@ -0,0 +1,355 @@
+          else
+            builder.tspan(x: x, dy: '1.0em') { builder << '<br/>' }
+          end
+        else
+          line = Loofah.fragment(line).scrub!(:strip).text.unicode_normalize
+
+          line_breaks = pack_up_string(line, ' ', font_size, text_box_width)
+
+          line_breaks.each do |row|
+            safe_message = Loofah.fragment(row).scrub!(:escape)
+            builder.tspan(x: x, dy: '1.0em') { builder << safe_message }
+          end
+
+          previous_line_was_text = true
+        end
+      end
+    end
+
+    annotation.add_child(builder.target!)
+
+    # Remove the <switch> tag
+    annotation.xpath('switch').remove
+  end
+
+  # Save new shapes.svg copy
+  File.open("#{@published_files}/shapes_modified.svg", 'w', 0o600) do |file|
+    file.write(whiteboard)
+  end
+end
+
+def measure_string(s, font_size)
+  # https://stackoverflow.com/a/4081370
+  # DejaVuSans, the default truefont of Debian, can be used here
+  # /usr/share/fonts/truetype/dejavu/DejaVuSans.ttf
+  # use ImageMagick to measure the string in pixels
+  command = "convert xc: -font /usr/share/fonts/truetype/msttcorefonts/Arial.ttf -pointsize #{font_size} -debug annotate -annotate 0 #{Shellwords.escape(s)} null: 2>&1"
+  _, output = run_command(command, true)
+  output.match(/; width: (\d+);/)[1].to_f
+end
+
+def pack_up_string(s, separator, font_size, text_box_width)
+  # Split the line on whitespaces, and measure the line to fit into the text_box_width
+  line_breaks = []
+  queued_words = []
+  s.split(separator).each do |word|
+    # First consider queued word and the current word in the line
+    test_string = (queued_words + [word]).join(separator)
+
+    width = measure_string(test_string, font_size)
+
+    if width > text_box_width
+      # Line exceeded, so consider the queued words as a line break and queue the current word
+      line_breaks += [queued_words.join(separator)]
+      if measure_string(word, font_size) > text_box_width
+        # If the word alone exceeds the box width, then we pack the word maximizing the amount of characters on each line
+        res = pack_up_string(word, '', font_size, text_box_width)
+        # Queue last line break, other words might fit
+        queued_words = [res.pop]
+        line_breaks += res
+      else
+        queued_words = [word]
+      end
+    else
+      # Current word fits the text box, so keep enqueueing new words
+      queued_words += [word]
+    end
+  end
+  # Make sure we release the final queued words as the final line break
+  line_breaks += [queued_words.join(separator)] unless queued_words.empty?
+
+  line_breaks
+end
+
+def parse_whiteboard_shapes(shape_reader)
+  slide_in = 0
+  slide_out = 0
+
+  shapes = []
+  slides = []
+
+  shape_reader.each do |node|
+    next unless node.node_type == Nokogiri::XML::Reader::TYPE_ELEMENT
+
+    node_name = node.name
+    node_class = node.attribute('class')
+
+    if node_name == 'image' && node_class == 'slide'
```

Comment:
Found an issue when the meeting is started with `autoStartRecording`, an empty blank first slide is generated because the presentation is not loaded yet, think you just need to ignore the slide with href `presentation/logo.png` here based on the the shapes_modififed.svg:
`<image id="image1" class="slide" in="0.0" out="3.0" href="presentation/logo.png" width="1600" height="1200" x="0" y="0" style="visibility:hidden"/>`